### DISCUSSION TITLE
Checkboxes using HTML appear on Web Viewer, but dissapear when exported to PDF

### DISCUSSION THREAD

**Discussion by stevehydro** [eclipse-birt/birt#1548] [Q&A]
Hi,

The code I use is a variable with the following code:

_**if(params["DisplayPurchasePrice"].value && params["DisplayPurchasePrice"].value == "Yes"){
	" & # x2611;";
} else {
	" & # x2612;";
}**_

(note: i had to put spaces in the 2 output symbols for them to appear on the forum)

for the field on the report, i have the code:

_var label = "Display Purchase Price: ";
var value = vars["displayPurchasePrice"];
label.bold()+value;_

When I run it for Web Viewer, a checkmark appears, but when exported to PDF, it disappears. 

Any way to fix this or do it another way?

Version: Neon.3 Release (4.6.3)
Build id: 20170314-1500

**Comment by speckyspooky** [comment]
Could your provide a short example of your report without external dataset.
It would be better to understand what your topic is. I didn't understand it in full.

Have you tested your report on a later version because 4.6 is an older one.

**Comment by hvbtup** [comment]
This is probably a font issue. You need a TrueType font which supports those characters.
Besides, you should try the same without using HTML entities. You can use unicode literals or (in literal strings only) the \u escape syntax.

**Reply by stevehydro** [reply]
how do i use unicode?

it only disappears when exported to PDF

i found the checkbox to be U+2612, i put that in, but it just displayed the text.

if(params["DisplayPurchasePrice"].value && params["DisplayPurchasePrice"].value == "Yes"){
"U+2612";
} else {
"U+2610";
}

**Reply by hvbtup** [reply]
You can use unicode instide string literals in Javascript like this: `"\u2610"`
The same syntax works for Java string literals, btw.

**Comment by speckyspooky** [comment]
You can add unicode signs with java-functions based on your example:

`java.lang.String.valueOf(java.lang.Character.toChars(java.lang.Integer.parseInt("2612", 16)));`

But not all unicode signs will be display. Thefore it can make sense according to @hvbtup comment to use a specific symbole TrueType font.
Below I added a screen and the according demo-report. May be it will help you on your case.

### Screens of small mockup
![2024-01-27 19_31_29-simple report parameter test - icon_central_symbol pdf – Mozilla Firefox](https://github.com/eclipse-birt/birt/assets/41593722/747b620e-0070-4640-9c72-b154dd6ba175)

**Reply by speckyspooky** [reply]
Two additional hints / ideas:

- idea 01: with BIRT 4.14 there is a new option implemented where you are able to create diagonal lines in a grid and if you have a grid 1x1 you can use the diagonals for the X-box

- idea 02: you can create with BIRT 4.6 a grid with size 1x1 and print the character "X" on it in an according style/font which you need to simulate the checked box

- idea 03: you can use images on float text based on text with type "HTML" and embedded images for the checkboxes

By the way, the missing unicode-character on PDF-side is a PDF-topic of openpdf and at the and cuased through licence-topics of fonts.
There are some of such dicussions on the openpdf-project.

### Demo-report based on font "wingdings" & "wingdings 2" and HTML-image-tag
[Uploading icon_symbol_test.zip…]()
